client/src/app/pages/migration-targets/migration-targets.tsx

```diff
@@ -28,37 +28,158 @@ import { useTranslation } from "react-i18next";
 
 import { Item } from "./components/dnd/item";
 import { DndGrid } from "./components/dnd/grid";
-import { useFetchMigrationTargets } from "@app/queries/rulesets";
-import { CustomTargetForm } from "./custom-target-form";
+import { BundleOrderSetting, RuleBundle, Setting } from "@app/api/models";
+import { updateBundleOrderSetting } from "@app/api/rest";
+import { AxiosError, AxiosPromise, AxiosResponse } from "axios";
+import {
+  BundleOrderSettingKey,
+  useDeleteRuleBundleMutation,
+  useFetchBundleOrder,
+  useFetchRuleBundles,
+} from "@app/queries/rulebundles";
+import { useEntityModal } from "@app/shared/hooks/useEntityModal";
+import { NotificationsContext } from "@app/shared/notifications-context";
+import { getAxiosErrorMessage } from "@app/utils/utils";
+import { UpdateCustomTargetModal } from "./components/update-custom-target-modal/update-custom-target-modal";
+import { NewCustomTargetModal } from "./components/new-custom-target-modal";
 
 export const MigrationTargets: React.FC = () => {
   const { t } = useTranslation();
+  const { pushNotification } = React.useContext(NotificationsContext);
+
+  const {
+    ruleBundles,
+    isFetching: isFetchingRuleBundles,
+    refetch: refetchRuleBundles,
+  } = useFetchRuleBundles();
+
+  const {
+    bundleOrderSetting,
+    isFetching,
+    refetch: refreshBundleOrderSetting,
+  } = useFetchBundleOrder(ruleBundles);
 
-  const { migrationTargets } = useFetchMigrationTargets();
+  useEffect(() => {
+    refetchRuleBundles();
+    refreshBundleOrderSetting();
+  }, [isFetching]);
```

Comment:
Hmm, @ibolton336 is it possible for this `useEffect` to cause an infinite loop? If I'm reading it correctly it'll run the effect whenever `isFetching` changes, which is triggered by `refreshBundleOrderSetting` in the effect. If it's not a loop it may be running more times than intended (since that refresh call will switch `isFetching` to true and then to false, so the effect will run twice). If this is not causing a loop I don't actually understand why 😅 some magic in react-query?

What's the intended trigger for this effect? If you simply want to fetch something else when the `refreshBundleOrderSetting` is done, you may be better off passing in an `onSuccess` to the query within `useFetchBundleOrder`.